frontend/service.go

```diff
@@ -177,6 +201,35 @@ func (s *lwdStreamer) GetBlockRange(span *walletrpc.BlockRange, resp walletrpc.C
 	}
 }
 
+// GetBlockRangeNullifiers is the same as GetBlockRange except that only
+// the actions contain only nullifiers (a subset of the full compact block).
+func (s *lwdStreamer) GetBlockRangeNullifiers(span *walletrpc.BlockRange, resp walletrpc.CompactTxStreamer_GetBlockRangeNullifiersServer) error {
+	blockChan := make(chan *walletrpc.CompactBlock)
+	if span.Start == nil || span.End == nil {
+		return errors.New("must specify start and end heights")
+	}
+	errChan := make(chan error)
+	go common.GetBlockRange(s.cache, blockChan, errChan, int(span.Start.Height), int(span.End.Height))
+
+	for {
+		select {
+		case err := <-errChan:
+			// this will also catch context.DeadlineExceeded from the timeout
+			return err
+		case cBlock := <-blockChan:
+			err := resp.Send(cBlock)
+			for _, tx := range cBlock.Vtx {
+				for i, action := range tx.Actions {
+					tx.Actions[i] = &walletrpc.CompactOrchardAction{Nullifier: action.Nullifier}
```

Comment:
same as above. Shouldn't Sapling nullifiers be returned as well?

Reply:
Oh cool! thanks I didn't understand it directly from the code above because I was lacking the context.